component/src/main/java/io/siddhi/extension/io/file/FileHandler.java

```diff
@@ -0,0 +1,409 @@
+                                "last_modified string, status string);\n" +
+                                "@sink(type='log')" +
+                                "define stream FooStream (filepath string, length int, " +
+                                "last_modified string, status string); \n",
+
+                        description = "" +
+                                "Under above configuration, all the files in the specific uri " +
+                                "which has been  modified or created or deleted in the execution time " +
+                                "will be identified and displayed.\n After monitoring has been started, " +
+                                "If any files in the specific uri is newly created, modified or deleted.\n " +
+                                "an event is created with the filepath, length, last modified date " +
+                                "and status of the file. Then that will be received by the FooStream..\n"
+                ),
+                @Example(
+                        syntax = "" +
+                                "@source(type='fileeventlistener',uri='file://abc/xyz', monitoring.interval='200', " +
+                                "@map(type='json')) \n" +
+                                "define stream CreateFileStream (filepath string, length long, " +
+                                "last_modified string, status string);\n" +
+                                "@sink(type='log')" +
+                                "define stream FooStream (filepath string, length int, " +
+                                "last_modified string, status string); \n",
+
+                        description = "" +
+                                "Under above configuration, all the files in the specific uri \n" +
+                                "which has been  modified or created or deleted in the execution time " +
+                                "will be identified and displayed. \n After monitoring has been started," +
+                                " If any files in the specific uri is newly created, modified or deleted.\n " +
+                                "an event is created with the filepath, length, last modified date " +
+                                "and status of the file. Then that will be received by the FooStream. If there " +
+                                "are any changes a new event will be generated in every 200 milliseconds.\n"
+                ),
+
+        }
+)
+
+public class FileHandler extends Source<FileHandler.FileHandlerState> {
+    private static final Logger log = Logger.getLogger(FileHandler.class);
+    private SourceEventListener sourceEventListener;
+    private long monitoringInterval = 500;
+    private String uri;
+    private String listeningDirUri;
+    private String listeningFileUri;
+    private FileAlterationMonitor monitor;
+    private Map<String, Long> initialMap = new ConcurrentHashMap<>();
```

Comment:
lets rename 'initialMap' this to 'recentFileSizeMap'

Reply:
lets rename 'initialMap' this to 'recentFileSizeMap', since the 'initialMap' gets changed with time

Reply:
fixed in 3f9a2a0